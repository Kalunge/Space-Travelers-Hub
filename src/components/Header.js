import React from 'react';
import { NavLink, BrowserRouter as Router } from 'react-router-dom';
import logo from '../images/planet.png';

const Header = () => (
  <Router>
    <div className="d-flex justify-content-between pt-3 align-items-center pl-3">
      <ul className="nav">
        <li className="nav-item">
          <img src={logo} alt="logo" className="navLogo" />
        </li>
        <li className="nav-item">
          <NavLink exact to="/" className="nav-link link-dark fs-4 fw-bolder">
            Space Travellers Hub
          </NavLink>
        </li>
      </ul>
      <ul className="nav">
        <li className="nav-item">
          <u>
            <NavLink to="/" className="nav-link">
              Rockets
            </NavLink>
          </u>
        </li>
        <li className="nav-item">
          <u>
            <NavLink to="/" className="nav-link border-end">
              Mission
            </NavLink>
          </u>
        </li>
        <li className="nav-item">
          <u>
            <NavLink to="/profile" className="nav-link">
              My profile
            </NavLink>
          </u>
        </li>
      </ul>
    </div>
  </Router>
);

export default Header;
