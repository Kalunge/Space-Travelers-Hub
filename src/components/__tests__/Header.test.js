import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Header from '../Header';

describe('snapshot test for Header Component', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Header />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
