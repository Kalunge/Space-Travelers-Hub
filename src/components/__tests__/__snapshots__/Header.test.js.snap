// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for Header Component renders correctly 1`] = `
Array [
  <div
    className="d-flex justify-content-between pt-3 align-items-center pl-3 mb-5"
  >
    <ul
      className="nav"
    >
      <li
        className="nav-item"
      >
        <img
          alt="logo"
          className="navLogo"
          src="planet.png"
        />
      </li>
      <li
        className="nav-item"
      >
        <a
          aria-current="page"
          className="nav-link link-dark fs-4 fw-bolder active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Space Travellers Hub
        </a>
      </li>
    </ul>
    <ul
      className="nav"
    >
      <li
        className="nav-item"
      >
        <a
          aria-current="page"
          className="nav-link active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Rockets
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          aria-current={null}
          className="nav-link missions"
          href="/missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          aria-current={null}
          className="nav-link"
          href="/profile"
          onClick={[Function]}
        >
          My profile
        </a>
      </li>
    </ul>
  </div>,
  <div
    className="boorder mb-2"
  />,
]
`;
